gw_eccentricity/eccDefinition.py

```diff
@@ -192,12 +234,15 @@ def __init__(self, dataDict, num_orbits_to_exclude_before_merger=2,
         if not np.allclose(self.t_diff, self.t_diff[0]):
             raise Exception("Input time array must have uniform time steps.\n"
                             f"Time steps are {self.t_diff}")
-        self.hlm = self.dataDict["hlm"]
-        self.h22 = self.hlm[(2, 2)]
-        self.amp22 = np.abs(self.h22)
-        self.phase22 = - np.unwrap(np.angle(self.h22))
-        self.omega22 = time_deriv_4thOrder(self.phase22,
-                                           self.t[1] - self.t[0])
+        # get phase22 from dataDict
+        self.phase22, self.omega22, self.h22 = self.compute_phase22(self.dataDict)
```

Comment:
Well, it does more than compute phase...
Maybe a single function called decompose_waveform, which returns phase22, omega22, amp22 and h22?

Reply:
I agree with using a better name. But putting everything in the same function makes it too long. Also, we do not need to compute each of these data at the same time. For example, in the truncate_dataDict.., we just need the phase and amp, we do not need h22 or omega22 there.